### PR TITLE
Preliminary implementation of Selected Post Summary Chart View

### LINKED ISSUE #11062: [Stats Refresh] Selected Post : Construct Chart for Selected Post Performance
A user can visualize the performance of a selected post, as measured by Views. The chart should allow the user to accomplish the following:

- Show a bar graph of views with available data.
- Adjust chart scale and dates relevant to data displayed

In many ways, this chart is similar to the [LPS chart](https://github.com/wordpress-mobile/WordPress-iOS/issues/11061), including the same stubbed data, with the following additional functionality:

- Graphically highlight the selected period on the chart.
- Support tapping on another bar to refresh the screen for the newly selected period.

### PR DESCRIPTION
**Pull request by stevebaranski**
Fixes #11062, which calls for the creation of a chart for a selected post. We don't yet have a _Post Details_ screen, so I'm using the _Latest Post Summary_ to scaffold and demonstrate these changes.

To test:

- Checkout the branch, confirm that it builds, and that existing tests pass.
- Manually run the app and visually confirm that a site with a recent post still looks just like the video in #11109.
- Now manually revert commit `7d28d9f`, run the app, and confirm that it looks a bit more like _Exhibit A_ below.

Update release notes:

- [X] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

<details>
<summary>Exhibit A</summary>

11062a_pr

</details>

### CODE REVIEW

**Comment by stevebaranski** [comment]
@frosty in the absence of both Stephenie & Jan, would you have some spare cycles to review this please?

File: WordPress/Classes/ViewRelated/Stats/Charts/Charts+Support.swift [hunk 1]
```diff
@@ -18,6 +18,8 @@ extension BarChartDataSet {
 
 extension NSUIColor {
     static let chartColor = UIColor(red: 135/255.0, green: 166/255.0, blue: 188/255.0, alpha: 255.0/255.0)
+
+    static let highlightColor = UIColor(red: 245/255.0, green: 131/255.0, blue: 53/255.0, alpha: 255.0/255.0)
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Stats/Charts/Charts+Support.swift]: Should we refer to named colors from `WPStyleGuide`?

File: WordPress/Classes/ViewRelated/Stats/Charts/StatsBarChartView.swift [hunk 2]
```diff
@@ -110,10 +109,17 @@ class StatsBarChartView: BarChartView {
 
         if let dataSets = barChartData.dataSets as? [BarChartDataSet] {
             for dataSet in dataSets {
+                dataSet.colors = [ styling.barColor ]
+
                 dataSet.drawValuesEnabled = false
-                dataSet.highlightEnabled = false
 
-                dataSet.colors = [ styling.barColor ]
+                if let barHilightColor = styling.highlightColor {
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Stats/Charts/StatsBarChartView.swift]: Nitpick `barHilightColor` -> `barHighlightColor`?

Comment (stevebaranski) [reply WordPress/Classes/ViewRelated/Stats/Charts/StatsBarChartView.swift]: I will take care of this at once!

File: WordPress/Classes/ViewRelated/Stats/Charts/Charts+Support.swift (hunk 1 above)
Comment (stevebaranski) [reply WordPress/Classes/ViewRelated/Stats/Charts/Charts+Support.swift]: The colors were obtained from Zeplin; I was unable to find exact matches in the style guide. There were a couple of considerations that influenced this approach:

- The type (i.e., `NSUIColor`).
- The `chartColor` above is not part of the style guide, but is used [elsewhere](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/ViewRelated/Reader/WPStyleGuide%2BReader.swift#L55-L58) via a Reader-specific extension.

As for the highlight color here, I'll confer with @SylvesterWilmott about alternatives.

Comment (stevebaranski) [reply WordPress/Classes/ViewRelated/Stats/Charts/Charts+Support.swift]: I have updated the colors based on discussion via `p1551456028012400-slack-nde-stats-refresh`.

**Comment by stevebaranski** [comment]
@frosty thanks for the review. I have pushed two commits that address your feedback. I believe this is ready for another look.

**Comment by stevebaranski** [comment]
@frosty thanks for bringing up the issue you observed with legibility. I have pushed two additional commits that resolve the outstanding issues:

1. Apply two separate colors for the axes & labels, as agreed upon via `p1551716015016600-slack-nde-stats-refresh`.
1. Addressed the rotation issue you noted previously. 

I've included a GIF that demonstrates both of these things; capturing it on device isn't ideal due to rotation, but hopefully it helps get the point across.

11062pr_feedback

Mind taking another look?

**Comment by SylvesterWilmott** [comment]
Looks great @stevebaranski ! The colours on the chart are spot on now.

I understand this is a WIP but just to clarify, the user should not be able to tap any of the bars in the chart on the Latest Post Summary Block. This means there is no selected bar state for this block and they should all be blue.

Also we have a new layout for Latest Post Summary block, is this something we should address in this PR?

**Comment by stevebaranski** [comment]
Thanks for the feedback @SylvesterWilmott! I'd like to respond to each comment in turn.

> I  understand this is a WIP but just to clarify, the user should not be able to tap any of the bars in the chart on the Latest Post Summary Block. This means there is no selected bar state for this block and they should all be blue.

Right. I tried to call this out in the PR. "We don't yet have a Post Details screen, so I'm using the Latest Post Summary to scaffold and demonstrate these changes." So this PR uses the LPS card as a harness for the changes, then promptly replaces the injected selected-post back to the LPS appearance. Once merged, the LPS summary will look the same as it did before. 

We'll need to amend #11068 to reflect that it is blocked by the creation of that screen. I need to confer with @ScoutHarris about what issue she has in mind to resolve that.

> Also we have a new layout for Latest Post Summary block, is this something we should address in this PR?

My work on this project is limited to the Charts themselves. I don't see a specific issue created for modifying the appearance of the LPS card, but it is noted in the Master Issue here:

screen shot 2019-03-05 at 6 38 52 am

File: WordPress/Classes/ViewRelated/Stats/Charts/StatsBarChartView.swift [hunk 3]
```diff
@@ -135,6 +178,19 @@ class StatsBarChartView: BarChartView {
...
     }
+
+    private func redrawChartMarkersIfNeeded() {
+        guard marker != nil, let highlight = lastHighlighted, let entry = barData?.entryForHighlight(highlight) else {
+            return
+        }
+
+        notifyDataSetChanged()
+
+        let postRotationDelay = DispatchTime.now() + TimeInterval(0.3)
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Stats/Charts/StatsBarChartView.swift]: Is this just required so that it happens after the rotation is complete?

Comment (stevebaranski) [reply WordPress/Classes/ViewRelated/Stats/Charts/StatsBarChartView.swift]: `notifyDataSetChanged` redraws the chart post-rotation, but the framework does not necessarily understand how the custom marker view should be redrawn in response.

Without a delay, I observed that the call from `getBarBounds(entry:)` was incorrect, and it wasn't rendering correctly.

**Comment by stevebaranski** [comment]
Thanks for the review @frosty!

**Pull request merged**